python/taichi/lang/impl.py

```diff
@@ -202,19 +202,24 @@ def create_program(self):
     def materialize(self):
         if self.materialized:
             return
+
+        print('[Taichi] materializing...')
         self.create_program()
 
         def layout():
             for func in self.layout_functions:
                 func()
 
         import taichi as ti
-        ti.trace("Materializing layout...".format())
+        ti.trace('Materializing layout...')
         taichi_lang_core.layout(layout)
         self.materialized = True
         for var in self.global_vars:
-            assert var.ptr.snode(
-            ) is not None, 'Some variable(s) are not placed'
+            if var.ptr.snode() is None:
+                raise RuntimeError(
+                    'Some variable(s) are not placed.'
+                    ' Did you forget to specify shape for a field? e.g.:\n'
+                    '  ti.field(ti.f32, shape=(3, 4))')
```

Comment:
I think `ti.field(ti.f32, shape=(3, 4))` is better since it empersize the `shape=`, wdyt?

Reply:
That makes sense, but should we always use keyword arguments in `ti.field` so that newcomers have better knowledge on which argument is which? (given there are quite a few possible arguments there...)